src/lightning/pytorch/trainer/trainer.py

```diff
@@ -286,6 +291,76 @@ def __init__(
         if default_root_dir is not None:
             default_root_dir = os.fspath(default_root_dir)
 
+        self.barebones = barebones
+        if barebones:
+            # opt-outs
+            if enable_checkpointing:
+                raise ValueError(
+                    f"`Trainer(barebones=True, enable_checkpointing={enable_checkpointing!r})` was passed."
+                    " Checkpointing can impact performance so it is disabled in barebones mode."
+                )
+            enable_checkpointing = False
+            if logger is not None and logger is not False:
```

Comment:
does this also disable the internal handling of logged values (and their aggregation) or just the flushing to the logger?

Reply:
Yes it does. See https://github.com/Lightning-AI/lightning/pull/16854/files#diff-4fdfbdac03762c6e625f1289e98750edd4c585a471e049a02f41d56922bf8be9R392-R397